src/Handlers/NetworkTestHandler.ts

```diff
@@ -43,8 +43,19 @@ export class NetworkTestHandler
 
         stdout.push(...lines);
 
-        if (chunk.indexOf(ReadlinePlatform.URLS_QUESTION) > -1) {
-          child.stdin.write(`${urls.join(',')}${EOL}`);
+        if (
+          chunk.indexOf(ReadlinePlatform.URLS_QUESTION) > -1 &&
+          config.type === NetworkTestType.PING
+        ) {
+          child.stdin.write(`${config.urls.join(',')}${EOL}`);
+        }
+
+        if (
+          chunk.indexOf(ReadlinePlatform.HOST_OR_IP_QUESTION) > -1 &&
+          config.type === NetworkTestType.TRACEROUTE
+        ) {
+          child.stdin.write(`${config.url}${EOL}`);
+          resolve(this.processOutput(stdout));
```

Comment:
1. There is a `resolve` on `:58` but no `resolve` on `:51`, is this intended?
2. I'd suggest to reverse this logic i.e. to check for `config.type` first and then to do something with `chunk`